src/coreclr/jit/instr.cpp

```diff
@@ -1096,19 +1175,64 @@ void CodeGen::inst_RV_TT_IV(instruction ins, emitAttr attr, regNumber reg1, GenT
     }
 }
 
+//------------------------------------------------------------------------
+// IsEmbeddedBroadcastEnabled: determine if embedded broadcast can be enabled
+//
+// Arguments:
+//    ins       -- The instruction being emitted
+//    op        -- The second operand of the instruction.
+//
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+bool CodeGenInterface::IsEmbeddedBroadcastEnabled(instruction ins, GenTree* op)
+{
```

Comment:
It would be good to early exit with `GetEmitter()->UseEvexEncoding()` first. A lot of user hardware may not have EVEX at all and we can avoid many of these checks in that case.